local/rest_api_gcbm/gcbm.py

```diff
@@ -226,25 +247,50 @@ def add_file(self, file_path: str):
         Parameters
         ==========
         1. file_path (str), no default
+        2. category (str), default = "", if skipped - then the categories will be deduced from the file path 
         """
 
         # TODO: update to accept input from Flask endpoint
+        # FIXME: The flask end point should do the pre-processing to be able to only pass the `file_path` as a string
+        # TODO: update in app.py to send valid data to add_file
         filename = os.path.basename(file_path)
         shutil.copy(file_path, os.path.join(self.dirpath, filename))
 
-        if self.disturbances._append(filename):
-            self.disturbances._update_config()
-            return
-        if self.classifiers._append(filename):
-            self.classifiers._update_config()
-            return
-        if self.miscellaneous._append(filename):
-            self.miscellaneous._update_config()
-            return
-        # TODO: Add covariates here
+        def _disturbance(filename, check=True):
+            if self.disturbances._append(filename, check):
+                self.disturbances._update_config()
+                
+        def _classifier(filename, check=True):
+            if self.classifiers._append(filename, check):
+                self.classifiers._update_config()
+
+        def _miscellaneous(filename, check=True):
+            if self.miscellaneous._append(filename, check):
+                self.miscellaneous._update_config()
+
+        if category != "":
+            if category == "disturbances":
+                _disturbance(filename, check=False)
+            elif category == "classifiers":
+                _classifier(filename, check=False)
+            elif category == "miscellaneous":
+                _miscellaneous(filename, check=False)
+            else:
+                # We can also raise an error here
+                raise UserWarning(f"Given category {category} not supported, supported categories are: {disturbances, classifiers, miscellaneous}")
+        else:
+            print(f"Category wasn't provided, attempting to deduce it from the file path: {file_path}")
```

Comment:
Another scenario is that categories are provided as part of an HTTP request, via call to the `/gcbm/upload` REST endpoint.

@YashKandalkar - would you prefer to support individual file uploads on the front end, or the bulk `upload` option we currently use an an example (see: [`curl.md`](https://github.com/moja-global/FLINT.Cloud/blob/master/local/rest_api_gcbm/curl.md))